app/Models/User.php

```diff
@@ -609,4 +607,52 @@ public function scopeTracked(Builder $query): Builder
     {
         return $query->where('Untracked', false); // TODO: use unranked_at=NULL?
     }
+
+    /**
+     * @param Builder<User> $query
+     * @return Builder<User>
+     */
+    public function scopeSearch(Builder $query, string $keyword): Builder
+    {
+        // Start with base conditions that apply to all searches.
+        $baseQuery = $query
+            ->whereNotNull('email_verified_at')
+            ->whereNull('Deleted')
+            ->whereNull('banned_at');
+
+        // Return early if keyword is empty to avoid unnecessary processing.
+        if (empty($keyword)) {
+            return $baseQuery;
+        }
+
+        // Common ranking calculation for both queries.
+        $activityRank = "
+            CASE
+                WHEN achievements_unlocked > 0 THEN 40
+                WHEN LastLogin > DATE_SUB(NOW(), INTERVAL 30 DAY) THEN 20
+                WHEN LastLogin > DATE_SUB(NOW(), INTERVAL 90 DAY) THEN 10
+                ELSE 0
```

Comment:
Should LastLogin be given more weight? It seems like I'd be more likely to be searching for an active user than one with a lot of points.

Reply:
I agree. I've made some adjustments in latest so equally-weighted results will now be ordered by LastLogin.